x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/add_domain_logic.ts

```diff
@@ -0,0 +1,169 @@
+  onValidateDomain(
+    newValue: string,
+    newEntryPointValue: string
+  ): { newValue: string; newEntryPointValue: string };
+  submitNewDomain(): void;
+  validateDomain(): void;
+}
+
+const DEFAULT_SELECTOR_VALUES = {
+  addDomainFormInputValue: 'https://',
+  entryPointValue: '/',
+};
+
+export const AddDomainLogic = kea<MakeLogicType<AddDomainLogicValues, AddDomainLogicActions>>({
+  path: ['enterprise_search', 'app_search', 'crawler', 'add_domain'],
+  actions: () => ({
+    clearDomainFormInputValue: true,
+    setAddDomainFormInputValue: (newValue) => newValue,
+    onSubmitNewDomainSuccess: (domain) => ({ domain }),
+    onSubmitNewDomainError: (errors) => ({ errors }),
+    onValidateDomain: (newValue, newEntryPointValue) => ({
+      newValue,
+      newEntryPointValue,
+    }),
+    submitNewDomain: true,
+    validateDomain: true,
+  }),
+  reducers: () => ({
+    addDomainFormInputValue: [
+      DEFAULT_SELECTOR_VALUES.addDomainFormInputValue,
+      {
+        clearDomainFormInputValue: () => DEFAULT_SELECTOR_VALUES.addDomainFormInputValue,
+        setAddDomainFormInputValue: (_, newValue: string) => newValue,
+        onValidateDomain: (_, { newValue }: { newValue: string }) => newValue,
+      },
+    ],
+    entryPointValue: [
+      DEFAULT_SELECTOR_VALUES.entryPointValue,
+      {
+        clearDomainFormInputValue: () => DEFAULT_SELECTOR_VALUES.entryPointValue,
+        setAddDomainFormInputValue: () => DEFAULT_SELECTOR_VALUES.entryPointValue,
+        onValidateDomain: (_, { newEntryPointValue }) => newEntryPointValue,
+      },
+    ],
+    // TODO When 4-step validation is added this will become a selector as
+    // we'll use individual step results to determine whether this is true/false
+    hasValidationCompleted: [
+      false,
+      {
+        clearDomainFormInputValue: () => false,
+        setAddDomainFormInputValue: () => false,
+        onValidateDomain: () => true,
+      },
+    ],
+    errors: [
+      [],
+      {
+        clearDomainFormInputValue: () => [],
+        setAddDomainFormInputValue: () => [],
+        onValidateDomain: () => [],
+        submitNewDomain: () => [],
+        onSubmitNewDomainError: (_, { errors }) => errors,
+      },
+    ],
+  }),
+  selectors: ({ selectors }) => ({
+    // TODO include selectors.blockingFailures once 4-step validation is migrated
+    allowSubmit: [
+      () => [selectors.hasValidationCompleted], // should eventually also contain selectors.hasBlockingFailures when that is added
+      (hasValidationCompleted: boolean) => hasValidationCompleted, // && !hasBlockingFailures
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    onSubmitNewDomainSuccess: ({ domain }) => {
+      flashSuccessToast(
+        i18n.translate(
+          'xpack.enterpriseSearch.appSearch.crawler.domainsTable.action.add.successMessage',
+          {
+            defaultMessage: "Successfully added domain '{domainUrl}'",
```

Comment:
my only comment here is that since this is dynamic info from the user, it might get really absurdly long (e.g. Cloud URLs or just `someonehasareallylongwebsite.com`). I'd prefer to stick with a generic `Successfully added your domain` (similar to how we use a generic "Successfully added your engine")

Reply:
Hmmmmm let me think about this.  I do like being specific in messaging like this to the user.  If I wanted to keep this level of information would it be better to move this to a Callout flash message?

Reply:
I think we should leave it in a toast to be consistent with the rest of our app's success messages.

I also want to note this is an EUI guideline: https://elastic.github.io/eui/#/display/toast/guidelines, specifically

> Do. Include the object name if it's not too long. Use single quotation marks around the object name if it helps clarify meaning.

One option is we use a JS snippet to truncate past a certain character length, or EUI may already be truncating toasts - I'd try a really long URL just to be sure nothing wonky happens